lib/admin/index.js

```diff
@@ -1,13 +1,15 @@
 // TODO: this is EG admin client; internal utility
-module.exports = function (adminConfig) {
-  let host = adminConfig.hostname || 'localhost';
-  let port = adminConfig.port || 9876;
-  adminConfig.baseUrl = adminConfig.baseUrl || `http://${host}:${port}/`;
+module.exports = function ({cliConfig, headers, verbose}) {
+  let host = cliConfig.hostname || 'localhost';
+  let port = cliConfig.port || 9876;
+  let protocol = cliConfig.protocol || 'http';
+  let baseUrl = cliConfig.baseUrl || `${protocol}://${host}:${port}/`;
```

Comment:
Could we just do a single `url` parameter in the `cliConfig` instead of `protocol`, `host`, and `port`?